compiler/sigmatch.nim

```diff
@@ -1650,7 +1650,7 @@ proc matchesAux(c: PContext, n, nOrig: PNode,
     if a >= formalLen-1 and formal != nil and formal.typ.isVarargsUntyped:
       incl(marker, formal.position)
       if container.isNil:
-        container = newNodeIT(nkBracket, n.sons[a].info, arrayConstr(c, n.info))
+        container = newNodeIT(nkArgList, n.sons[a].info, arrayConstr(c, n.info))
```

Comment:
I think the nkArgList solution is quite nice, but it needs to be documented as a breaking change in the news for 0.17.0. Also nkArgList is not handled in the codegens etc, they only know about nkBracket. I'm confused why this doesn't cause more code changes and breaking tests...

> it's unlikely that there will be code, which checks the node kind of the macro param - most of the code probably just takes the len and iterates over the node's children.

That's just good defensive macro programming though. This change needs to be documented.